### PR TITLE
Tweaks to heirlooms

### PR DESCRIPTION
**Pull request by Rukofamicom**
## About The Pull Request
* Removes wearable heirlooms entirely
* Re-organizes heirloom code logic
  * First is a 33% chance at job-specific heirloom
  * Then a 33% chance at racial heirloom (using prob(50) because of how compound probability works)
  * Then finally a generic heirloom is assigned if the first two failed, functioning as both the final 33% and also a failsafe
* Adds lizard and IPC racial heirlooms
* Adds generic heirlooms (key and random coin) that anyone may get regardless of race/job
* Removes D20 and Deck of cards heirlooms as generic options, now that generic will be popping up a lot more often. These didn't seem particularly heirloom-y to me, and a deck of cards has a very real chance of being accidentally destroyed just by dealing all of the cards out of it. 

## Why It's Good For The Game
Free or almost free negatives just encourage gaming the quirk system instead of using it to build a character. Likewise these changes help to smooth the curve out at least somewhat so that certain jobs/races are not consistently getting heirlooms of a different size and impact than others. There is still going to be some disparity, but there should be less of it now. 

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

image

image

image

image

</details>

## Changelog
:cl:
tweak: Adjusted the pool of heirlooms increase variety of the type of heirloom you will receive, there is now a 33% chance to get a job heirloom, 33% chance to get a species heirloom (if your species has one) and a 33% chance to get a generic heirloom not related to job/race.
del: Removed all wearable family heirlooms, as well as D20 and deck of cards from the failsafe. 
add: Added random coin and key to generic heirlooms
add: Added racial heirlooms to lizard and IPC
tweak: Various job heirlooms have been shuffled and adjusted
code: All items that only exist as heirlooms are now in the same place as the rest of heirloom code instead of scattered through files. Heirloom logic was also re-organized. 
/:cl:

### CODE REVIEW

**Comment by Rukofamicom** [comment]
While this PR is open I am open to new suggestions for heirloom additions too, so long as they are normal, small or small in size.

**Comment by Rukofamicom** [comment]
Additionally this PR was opened in direct response to #11665

**Comment by MarkusLarsson421** [comment]
Perhaps these heirlooms could be converted to specifically not be wearable?

Alternatively massively reduce their chance of being selected to not be ideal for gaming the system. Feels a shame to outright remove them as having a bucket heirloom has its charm I think.

**Comment by Rukofamicom** [comment]
> as having a bucket heirloom has its charm I think.

... fuck, I forgot this was wearable.

**Comment by Programs-The-Station** [comment]
Hallelujah, I hated the Engineering Hardhat heirloom.

For other heirlooms:
Lizard Heirlooms:
1. Skull (Crusher trophy, but inactive?)
Oozeling:
1. Bottle of (Old) Slime Ooze (just water, make the bottle the important bit)
IPC:
1. Experimental Firmware Disk (Corrupted, doesn't do anything)
Plasmaman:
1. Papa's Plasma Refilling pump (One gas pump, the atmos item) 
Dionae: 
1. Random Seed pack
Psyphoza:
1. A random maint loot object that has the same aura as their family

Anyone:
A Random Coin or gem
Family photo (not sure if there's a fake photo that doesn't pop up an image when used, but that would be an easy button) 
Family photo album
Key for an apartment/home back at central

**Comment by ClownMoff** [comment]
This should be in draft then

**Comment by MarkusLarsson421** [comment]
> ... fuck, I forgot this was wearable.

Then again, it has it's charm. Outright removal because of fear of gamers isn't the right kind of justification for such a change I think. Not to mention, not all wearables are ideal. A bucket helmet is only so good until you need another helmet.

**Comment by ClownMoff** [comment]
No I'm not mad you marked a PR that was a WIP as a WIP 

I just thought this was going to be a PR opened for a while you figured out the race heirlooms

**Comment by Rukofamicom** [comment]
@Programs-The-Station I love the lizard and IPC suggestion, I don't think I can do the others in an especially good way though. I've included the random coin and key as universal heirlooms too now.

**Comment by Rukofamicom** [comment]
Pending further suggestions this is ready again.

File: code/datums/traits/negative_quirk.dm [hunk 1]
```diff
@@ -218,11 +217,24 @@
...
+			heirloom_type = /obj/item/heirloom_skull
+		if((ismoth(H)))
+			heirloom_type = /obj/item/flashlight/lantern/heirloom_moth
+		if(isipc(H))
+			heirloom_type = /obj/item/disk/tech_disk
+
+
+	//Generic heirloom, anywhere from 33% chance to 100% chance depending on race/job combination.
+	// Functions as a failsafe in the event that no other heirlooms were available, that's why this isn't a switch.
 	if(!heirloom_type)
```
Comment (mystery3525) [inline code/datums/traits/negative_quirk.dm]: Shouldn't this be an `else if(prob(50))` and `else`? Looks like right now there's a 50%-flat chance to get a species-specific heirloom.

Comment (Rukofamicom) [reply code/datums/traits/negative_quirk.dm]: Yes, it should be. Thank you

**Comment by Fronsis** [comment]
Don't coins have a behaviour where they can be inserted into an ID for X amount of money? even if it's zero iirc you can still slap them in

**Comment by Rukofamicom** [comment]
Some parts of this can/will be salvaged, but not within this same PR. 

See my post here for my thoughts after getting feedback on this and the topic of traits as a whole: #11708

**Pull request closed without merging**